packages/routing-utils/src/regexr/expression-lexer.js

```diff
@@ -0,0 +1,952 @@
+/*
```

Comment:
Where did this come from? Is it copy/pasta? And if so, why?

Reply:
It's from regexr.com specifically this repo https://github.com/gskinner/regexr it's copied/pasted since it's not available through npm and we only need this specific lexer to gather the named groups from the regex correctly. 

Attempting to gather them without a lexer seems to be un-reliable and the same lexer is being added in Next.js' handling here https://github.com/vercel/next.js/pull/23626

Reply:
@ijjk What about `yarn add --dev https://github.com/gskinner/regexr` so that its added to the lock file?

Reply:
We would need to add a build step if we pull in the source code directly since it's using import/export syntax. We only need the expression lexer so seems better to only pull that in here instead of adding an additional dependency/build step.

I could create a separate repo with just the lexer with the source added here if that's preferable.

Reply:
I'm just thinking long term, how do we handle updating this code in case it changes upstream (think bugs, vulnerabilities, etc)?

Forking and publishing to npm won't really solve that problem. But I guess it will solve the problem of shared code between `vercel/vercel` and `vercel/next.js` if we need to modify regexr.

> since it's using import/export syntax

Node.js 12.17.0+ supports ESM without a flag so we could make this work without a build step I believe (not sure how TS will mix ESM with CJS though 🤔 )

Reply:
I updated to use a shared repo here https://github.com/ijjk/regexr-lexer that has a built version of the lexer/profiles included that is added to the lockfile. 

Do we want to ncc the routing-utils files as well or do we need to keep the same output structure?

Reply:
We don't need to use `ncc` here because that will happen downstream, either CLI or API